core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -51,59 +45,24 @@
   private static final Logger LOG = LoggerFactory.getLogger(LocalPageStore.class);
 
   private final String mRoot;
-  private final AtomicLong mSize = new AtomicLong(0);
-  private final AtomicLong mBytes = new AtomicLong(0);
   private final long mPageSize;
+  private final long mCacheSize;
   private final int mFileBuckets;
   private final Pattern mPagePattern;
 
   /**
    * Creates a new instance of {@link LocalPageStore}.
    *
    * @param options options for the local page store
-   * @throws IOException when fails to create a {@link LocalPageStore}
    */
-  public LocalPageStore(LocalPageStoreOptions options) throws IOException {
+  public LocalPageStore(LocalPageStoreOptions options) {
     mRoot = options.getRootDir();
     mPageSize = options.getPageSize();
+    mCacheSize = options.getCacheSize();
     mFileBuckets = options.getFileBuckets();
-    Path rootDir = Paths.get(mRoot);
     // pattern encoding root_path/page_size(ulong)/bucket(uint)/file_id(str)/page_idx(ulong)/
     mPagePattern = Pattern.compile(
-        String.format("%s/%d/(\\d+)/([^/]+)/(\\d+)", Pattern.quote(rootDir.toString()), mPageSize));
-    try {
-      boolean invalidPage = false;
-
-      if (Files.exists(rootDir)) {
-        try (Stream<Path> stream = Files.walk(rootDir)) {
-          invalidPage = stream
-              .filter(Files::isRegularFile)
-              .anyMatch(path -> {
-                if (getPageId(path) == null) {
-                  LOG.warn("Invalid page path {}", path);
-                  return true;
-                }
-                try {
-                  mBytes.getAndAdd(Files.size(path));
-                } catch (IOException e) {
-                  LOG.warn("Fail to get file size {}", e.toString());
-                }
-                mSize.incrementAndGet();
-                return false;
-              });
-        }
-      }
-
-      if (invalidPage || mBytes.get() > options.getCacheSize()) {
-        LOG.warn("Cannot recover from cached data: {}",
-            invalidPage ? "Invalid page file found" : "Cached data size exceeded configured value");
-        FileUtils.cleanDirectory(new File(mRoot));
-        mSize.set(0);
-        mBytes.set(0);
-      }
-    } catch (IOException e) {
-      throw new IOException(String.format("can't initialize page store at %s", mRoot), e);
-    }
+        String.format("%s/%d/(\\d+)/([^/]+)/(\\d+)", Pattern.quote(mRoot), mPageSize));
```

Comment:
It can be tricky for user to deal with trailing slash if we use the configured path without normalizing it.

Reply:
reverted  and add a javadoc preventing future change